packages/js/eslint-plugin/src/rules/key-name-casing.ts

```diff
@@ -0,0 +1,161 @@
+import { SourceCode } from "eslint"
+import { JSONProperty } from "jsonc-eslint-parser/lib/parser/ast"
+
+import { OpenverseRule } from "../utils/rule-creator"
+
+export type CasingKind = "camelCaseWithDot" | "snake_case_with_dot"
+
+export const allowedCaseOptions: CasingKind[] = [
+  "camelCaseWithDot",
+  "snake_case_with_dot",
+]
```

Comment:
Just a tip, if you want to avoid duplicating the names in this definition, you can derive `CasingKind` from `allowedCaseOptions`:

```suggestion
export const allowedCaseOptions = [
  "camelCaseWithDot",
  "snake_case_with_dot",
] as const

export type CasingKind = typeof allowedCaseOptions[number]
```

Alternatively, you can derive both from the `checkersMap`:

```typescript
const checkersMap = {
  camelCaseWithDot: isCamelCase,
  snake_case_with_dot: isSnakeCase,
} as const

export type CasingKind = keyof typeof checkersMap

export const allowedCaseOptions: CasingKind[] = Object.keys(checkersMap)
```